internal/db/domain.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"slices"
+	"strings"
+
+	"github.com/sqlc-dev/pqtype"
+)
+
+// This file contains domain-level methods for db structs
+
+// CanImplement returns true if the provider implements the given type.
+func (p *Provider) CanImplement(impl ProviderType) bool {
+	return slices.Contains(p.Implements, impl)
+}
+
+// ProfileRow is an interface row in the profiles table
```

Comment:
Moved across from `store.go` - I think it's best to separate out these domain-level methods and data types